eos_data_distribution/ndn/http.py

```diff
@@ -53,42 +55,52 @@ def read_bytes_async():
 
     read_bytes_async()
 
-def get_content_size(session, url):
+def fetch_http_headers(session, url):
     # XXX: SOMA's subscriptions-frontend doesn't handle HEAD requests yet because S3
     # is a bit silly with signed requests. For now, request a bytes=0-0 range and
-    # look at the Content-Range.
+    # return the full response_headers.
     msg = Soup.Message.new("GET", url)
     msg.request_headers.append('Range', 'bytes=0-0')
     session.send(msg, None)
-    content_range = msg.response_headers.get_one('Content-Range')
+    return msg.response_headers
+
+def get_content_size(headers):
+    content_range = headers.get_one('Content-Range')
     size = int(content_range.split('/')[1])
     return size
 
+def get_last_modified(headers):
+    # note that we can't use ETag as we need things to be ordered
+    date = Soup.Date.new_from_string(headers.get_one('Last-Modified'))
```

Comment:
What if there is no `Last-Modified` header? `headers.get_one()` will return `None` and `Date.new_from_string()` will probably be sad.